crates/ruma-client-api/src/error.rs

```diff
@@ -230,6 +230,31 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+impl From<FromHttpRequestError> for Error {
+    fn from(error: FromHttpRequestError) -> Self {
+        let kind = match error {
+            FromHttpRequestError::Deserialization(DeserializationError::Json(t)) => {
+                match t.is_syntax() {
+                    true => ErrorKind::NotJson,
+                    false => ErrorKind::BadJson,
+                }
```

Comment:
I just checked the `serde_json` docs again, seems more robust to match on `t.classify()` here, with `Io` getting an extra match arm (though it can't actually happen with the current deserialization code) and `Eof` being handled the same as `Syntax`.